### PR TITLE
Add remove method to route-repository

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | https://github.com/sulu/sulu-docs/pull/469

#### What's in this PR?

This PR add a method to remove the route with the repository.

#### BC Breaks/Deprecations

New method introduced to repository interface

#### To Do

- [x] Add breaking changes to UPGRADE.md
- [x] Create docs PR

### CODE REVIEW

**Comment by alexander-schranz** [comment]
Missing Test and Documentation

**Comment by alexander-schranz** [comment]
@danrot @chirimoya which types of features do we merge into master and which type we release in patch releases?

**Comment by wachterjohannes** [comment]
@alexander-schranz there is no testcase for the repository :) will add a docs PR

**Comment by danrot** [comment]
Currently we are releasing new "features" only if they have existed in Sulu 1.6 and we forgot to port them... But I think at some point we should find back to only fixing bugs in hotfix releases IMO. Since this seems to be a completely new feature I would tend to merge that into master instead of release/2.0.

**Comment by alexander-schranz** [comment]
@danrot @wachterjohannes would also prefer merging this into master instead of release/2.0

**Comment by wachterjohannes** [comment]
@alexander-schranz @danrot have rebased this PR

File: UPGRADE.md [hunk 1]
```diff
@@ -5,6 +5,10 @@
 When upgrading also have a look at the changes in the
 [sulu skeleton](https://github.com/sulu/skeleton/compare/2.0.2...2.0.3).
 
+### RouteRepositoryInterface changed
```
Comment (alexander-schranz) [inline UPGRADE.md]: need to be moved to `dev-master`

File: UPGRADE.md [hunk 2]
```diff
@@ -5,6 +5,10 @@
...
+
+In the `RouteRepositoryInterface` a new method was introduced `removed`.
```
Comment (danrot) [inline UPGRADE.md]: "a new `remove` method was introduced.

**Comment by danrot** [comment]
@alexander-schranz Can you please test/merge this PR?

File: src/Sulu/Bundle/RouteBundle/Tests/Functional/Entity/RouteRepositoryTest.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+use Sulu\Bundle\RouteBundle\Entity\Route;
+use Sulu\Bundle\RouteBundle\Entity\RouteRepositoryInterface;
+use Sulu\Bundle\TestBundle\Testing\SuluTestCase;
+
+class RouteRepositoryTest extends SuluTestCase
+{
+    public function testRemove(): void
+    {
+        $this->purgeDatabase();
```
Comment (danrot) [inline src/Sulu/Bundle/RouteBundle/Tests/Functional/Entity/RouteRepositoryTest.php]: Don't you want to move that to the `setUp` method already? That will be necessary as soon as the second test is implemented.

File: src/Sulu/Bundle/RouteBundle/Tests/Functional/Entity/RouteRepositoryTest.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        $entityManager = $this->getEntityManager();
+
+        /** @var RouteRepositoryInterface $repository */
+        $repository = $entityManager->getRepository(Route::class);
```
Comment (danrot) [inline src/Sulu/Bundle/RouteBundle/Tests/Functional/Entity/RouteRepositoryTest.php]: This initialization stuff could also be done in the `setUp` method, right?

**Comment by wachterjohannes** [comment]
@danrot @alexander-schranz fixed and ready to review again

**Pull request merged**